### ISSUE TITLE
smappe problem

### ISSUE THREAD

**Issue by matteos1** [apazga/smappee-domoticz-bash#5]
i've a problem a message error that not comunica with server
"The server encountered an unexpected condition which prevented it from fulfilling the request

**Comment by matteos1** [comment]
./smappee_plug_control.sh 0 8
{"success":"Logon successful!","header":"Logon to the monitor portal successful..."}{"errorMessage":"The server encountered an unexpected condition which prevented it from fulfilling the request"}

**Comment by apazga** [comment]
List your devices with `./smappee_plug_control.sh list` first to see if you are using the right one.

**Comment by matteos1** [comment]
I’ve seen the list with the above command, but I don’t manage to switch it on and off at the different switches. And when I give the command to switch on or off, the error message I receive is: “The server encountered an unexpected condition which prevented it from fulfilling the request.”

**Comment by apazga** [comment]
Can you show the list to see if is anything wrong?

Did you change the IP inside the script? By default is set to:
`SMAPPEE_IP="192.168.1.111"`

**Comment by matteos1** [comment]
Until today everything functioned well, and I installed Smappee a year ago. From today I cannot command it to switch on or off through Domoticz.

**Comment by apazga** [comment]
Ok, that's different. Let me check if mine is working. Maybe they updated or changed something
EDIT: Mine is working, nothing has changed, yet.
Did you try with another plug? Or just have one? To test if the plug is working

**Comment by matteos1** [comment]
all the plug not work
i have 6 plug

**Comment by matteos1** [comment]
Listing available plugs
Login...
{"success":"Logon successful!","header":"Logon to the monitor portal successful..."}

Available plugs:
[{"serialNumber":"N/A","type":"2","value":"Raspberry dietpikodi ","key":"4"},{"serialNumber":"N/A","type":"2","value":"Decoro corridoio ","key":"5"},{"serialNumber":"N/A","type":"2","value":"Raspberry plex ","key":"6"},{"serialNumber":"N/A","type":"2","value":"raspberry osmc ","key":"7"},{"serialNumber":"N/A","type":"2","value":"Pianta sotto caldaia ","key":"8"},{"serialNumber":"N/A","type":"2","value":"Luce piante ","key":"9"},{"serialNumber":"N/A","type":"2","value":"Raspberry  pi3 ","key":"13"},{"serialNumber":"N/A","type":"2","value":"Box musica ","key":"14"},{"serialNumber":"N/A","type":"2","value":"Amplificazione videoproiettore ","key":"15"},{"serialNumber":"N/A","type":"2","value":"Luce sateki ","key":"16"},{"serialNumber":"N/A","type":"2","value":"Videoproiettore cameretta ","key":"17"}]

**Comment by matteos1** [comment]
i meke an update to raspberry yesterday and udate this 
 libssl-dev libssl-doc libssl1.0.2 libssl1.1 openssl

**Comment by apazga** [comment]
Ok, that makes sense. Update curl & libcurl too.

**Comment by matteos1** [comment]
Curl is already update. Unable to locate libcurl

**Comment by apazga** [comment]
It may appear as libcurl3 (`dpkg -l | grep curl`)

**Comment by matteos1** [comment]
What is the command to update libcurl in raspberry ?

**Comment by matteos1** [comment]
apt-get install libcurl3
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libcurl3 is already the newest version.
libcurl3 set to manually installed.

**Comment by matteos1** [comment]
But not work :(

**Comment by apazga** [comment]
Ok, I'll try to update my libs to see if I can reproduce the issue.
Tell me your operating system & kernel version:
`uname -a`
`cat /etc/debian_version`

**Comment by matteos1** [comment]
Linux DietPi 4.4.36-v7+ #933 SMP Fri Dec 2 22:03:01 GMT 2016 armv7l GNU/Linux
8.0

**Comment by matteos1** [comment]
Some news?

**Comment by matteos1** [comment]
:(

**Comment by matteos1** [comment]
message error
{"errorMessage":"The server encountered an unexpected condition which prevented it from fulfilling the request"}

**Comment by apazga** [comment]
I updated everything but I can't reproduce your issue.
Tell me the specific version of that packages, please.
`dpkg -l | grep ssl`

**Comment by matteos1** [comment]
ii  libcurl4-openssl-dev:armhf                                       7.38.0-4+deb8u10                                          armhf        development files and documentation for libcurl (OpenSSL flavour)
ii  libflac8:armhf                                                   1.3.0-3                                                   armhf        Free Lossless Audio Codec - runtime C library
ii  libgnutls-openssl27:armhf                                        3.3.8-6+deb8u7                                            armhf        GNU TLS library - OpenSSL wrapper
ii  libssl-dev:armhf                                                 1.0.1t-1+deb8u8                                           armhf        Secure Sockets Layer toolkit - development files
ii  libssl-doc                                                       1.0.1t-1+deb8u8                                           all          Secure Sockets Layer toolkit - development documentation
ii  libssl1.0.0:armhf                                                1.0.1t-1+deb8u8                                           armhf        Secure Sockets Layer toolkit - shared libraries
ii  libwavpack1:armhf                                                4.70.0-1                                                  armhf        audio codec (lossy and lossless) - library
ii  openssl                                                          1.0.1t-1+deb8u8                                           armhf        Secure Sockets Layer toolkit - cryptographic utility
ii  python-openssl                                                   0.13.1-2                                                  armhf        Python 2 wrapper around the OpenSSL library
ii  ssl-cert                                                         1.0.35                                                    all          simple debconf wrapper for OpenSSL

**Comment by matteos1** [comment]
ok?

*matteo lioi*
*via udine 45*
*34135 trieste*
*tel. 3208505348*
*tel. 3496382492*

2018-04-06 23:02 GMT+02:00 Abel Paz <notifications@github.com>:

**Comment by matteos1** [comment]
​not work only for the plug. i see the other information regarding watt
consume etc etc​

**Comment by apazga** [comment]
I saw it's a common issue with new version of libssl, so you'll have to downgrade it:

```
wget http://ftp.nl.debian.org/debian/pool/main/o/openssl/libssl1.0.0_1.0.2l-1~bpo8+1_armhf.deb
sudo dpkg -i libssl1.0.0_1.0.2l-1~bpo8+1_armhf.deb
```

**Comment by matteos1** [comment]
Ok when I return at home I make this downgrade

**Comment by matteos1** [comment]
And I write here if is all ok thank you

**Comment by matteos1** [comment]
i make this upgrade but not work,,
there is always the same error
{"errorMessage":"The server encountered an unexpected condition which prevented it from fulfilling the request"}

**Comment by apazga** [comment]
Then, it's because of DietPi... sorry. If you can use Raspian instead, the issue will be gone.

**Comment by matteos1** [comment]
do you think that this is issue is for dietpi?
before this update all work ok..
now when i try to update libss there is other error message.

**Comment by apazga** [comment]
I tried different versions of Raspian, fully updated and is working.
Something is wrong with that update, may be is just a package version, but who knows...

**Comment by matteos1** [comment]
today i try to recover a copy of backup of dietpi when smappee plug work with domoticz.
where can i check the error and where are stored for this script?

**Comment by apazga** [comment]
There is no log info, comfort plugs are so simple.
You can check in your backup the library version that you have for libssl and related packages or dependencies that were updated too.

**Comment by matteos1** [comment]
I know that last update of dietpi has update libssl-dev libssl-doc libssl1.0.2 libssl1.1 openssl.
do you think that the problem is here?

**Comment by apazga** [comment]
Yes, maybe, there were many changes around SSL last months, and nothing was updated in Smappee

**Comment by matteos1** [comment]
i think that if a new ssl work in rasperry than it should work in dietpi cause have same library..

**Comment by matteos1** [comment]
i've restored my backup but always the same error
{"success":"Logon successful!","header":"Logon to the monitor portal successful..."}{"errorMessage":"The server encountered an unexpected condition which prevented it from fulfilling the request"}

**Comment by matteos1** [comment]
also on another raspberry give same error
list is this but when i plug on 
 ./smappee_plug_control.sh 1 9
{"success":"Logon successful!","header":"Logon to the monitor portal successful..."}{"errorMessage":"The server encountered an unexpected condition which prevented it from fulfilling the request"}root@raspberrypi:/home/pi/_scripts#

**Comment by matteos1** [comment]
do you have this version of software of smappee ??
cause till 29/03/2018 not work

Nr of features received from acquisition engine | 66890
-- | --
Last software update | 29/03/2018 12:15:39
Application software version and build nr | 1.0.0S buildNr #1185

**Comment by matteos1** [comment]
this is in var/tmp
lua: /usr/local/share/lua/5.2/JSON.lua:383: html passed to JSON:decode(): <html><head><title>Unauthorized</title></head><body><h1>401 Unauthorized</h1></body></html>
stack traceback:
	[C]: in function 'assert'
	/usr/local/share/lua/5.2/JSON.lua:383: in function 'onDecodeOfHTMLError'
	/usr/local/share/lua/5.2/JSON.lua:655: in function 'decode'
	/root/domoticz_smappee/smappee.lua:45: in function 'variable_list'
	/root/domoticz_smappee/smappee.lua:51: in function 'idx_from_variable_name'
	/root/domoticz_smappee/smappee.lua:108: in function 'retrieve_Domoticz_variable'
	/root/domoticz_smappee/smappee.lua:251: in main chunk
	[C]: in ?

**Comment by matteos1** [comment]
responce of smappee support
Dear,

It is indeed possible that the scripts will not work anymore. This is because the new firmware #1185 has changed the communication protocol used by the Smappee and the plugs. 

The older firmware used websockets as communication protocol. The protocol was highly unstable though at weak network connectivity. Therefor we have changed to an overall more stable communcation protocol: MQTT. 

You will have to change your scripts so that it will be compatible with this new **MQTT protocol**. 

I will now change the status of this ticket to "solved".

If you need further assistance, please do not hesitate to reply to this mail within 4 days. Off course you can always create a new ticket as well.

With best regards,

**Comment by apazga** [comment]
Oops, this is a big change. I'll try to update my Smappee, or wait for the update to update the scripts too.

**Comment by matteos1** [comment]
thank you

**Comment by matteos1** [comment]
Smappe support:
Dear,

Our development service has not yet uploaded the documentation which explains how to use the new MQTT protocol. Once this is uploaded, it will be uploaded to the same confluence page as the dev API. For now however this is not yet available. 
I have forwarded the scripts to my development team. They will look at the scripts and inform me how they need to be changed to use MQTT however this is not of their usual tasks. They have informed me it could possibly take some time. 

Once i have more information, i will definitely let you know.

**Comment by matteos1** [comment]
sorry, do you have some news?

**Comment by apazga** [comment]
Nope, no update for me yet. And Smappee will have to release their documentation to know how to use it.
I'll give a look and develop a new version this summer.

**Comment by matteos1** [comment]
here the new specific for MQTT protocol 
[Smappee local MQTT topics - 20180209 (1) (1).pdf](https://github.com/apazga/smappee-domoticz-bash/files/2000296/Smappee.local.MQTT.topics.-.20180209.1.1.pdf)
[MQTT broker configuration on Smappee Plus Pro (1) (2).pdf](https://github.com/apazga/smappee-domoticz-bash/files/2000297/MQTT.broker.configuration.on.Smappee.Plus.Pro.1.2.pdf)
